openedx/core/lib/api/tests/test_parsers.py

```diff
@@ -0,0 +1,142 @@
+            content_type='application/octet-stream',
+            HTTP_CONTENT_DISPOSITION='attachment; filename="VIRUS.EXE',
+        )
+        context = {'view': self.view, 'request': request}
+        result = self.parser.parse(
+            stream=BytesIO('abcdefgh'), media_type='application/octet-stream', parser_context=context
+        )
+        self.assertEqual(result.data, {})
+        self.assertIn('file', result.files)
+        self.assertEqual(result.files['file'].read(), 'abcdefgh')
+
+    def test_parse_mismatched_filename_and_mimetype(self):
+        """
+        Test that TypedFileUploadParser raises an exception when the specified
+        content-type doesn't match the filename extension in the
+        content-disposition header.
+        """
+        request = self.request_factory.post(
+            '/',
+            content_type='image/png',
+            HTTP_CONTENT_DISPOSITION='attachment; filename="file.jpg"',
+        )
+        context = {'view': self.view, 'request': request}
+        with self.assertRaises(exceptions.ParseError) as err:
+            self.parser.parse(stream=BytesIO('abcdefgh'), media_type='image/png', parser_context=context)
+            self.assertIn('developer_message', err.detail)
+            self.assertNotIn('user_message', err.detail)
+
+    def test_parse_any_type(self):
+        """
+        The view does not have to specify supported types.
+        """
+        view = object()
+        self.assertFalse(hasattr(view, 'upload_media_types'))
+
+        request = self.request_factory.post(
+            '/',
+            content_type='image/png',
+            HTTP_CONTENT_DISPOSITION='attachment; filename="file.png"',
+        )
+        context = {'view': view, 'request': request}
+        result = self.parser.parse(stream=BytesIO('abcdefgh'), media_type='image/png', parser_context=context)
+        self.assertEqual(result.data, {})
+        self.assertIn('file', result.files)
+        self.assertEqual(result.files['file'].read(), 'abcdefgh')
+
+    def test_parse_any_type_with_mismatched_extension(self):
+        """
+        When the view doesn't specify a set of supported types, the
+        TypedFileUploadParser will still check that the extension
+        and content-type match.
+        """
+        view = object()
+        self.assertFalse(hasattr(view, 'upload_media_types'))
+        request = self.request_factory.post(
+            '/',
+            content_type='image/png',
+            HTTP_CONTENT_DISPOSITION='attachment; filename="file.jpg"',
+        )
+        context = {'view': view, 'request': request}
+        with self.assertRaises(exceptions.ParseError):
+            self.parser.parse(stream=BytesIO('abcdefgh'), media_type='image/png', parser_context=context)
+
+    def test_parse_all_types_with_unknown_type(self):
+        """
+        If the view doesn't specify the set of supported types,
+        and the content-type is unknown, accept any file extension.
```

Comment:
By "strict" do you mean it should reject all requests (so that you have to specify upload_media_types to get it to work at all) or that it should only allow through requests for types that have defined extensions?

I could go either way, though I think I prefer the first: If you're going to use this parser, you should specify the types you accept.

Other opinions @nasthagiri @BenjiLee ?